### ISSUE TITLE
Fix ssl_password_file.t test

### ISSUE THREAD

**Issue by travisspencer** [curityio/nginx_phantom_token_module#32]
The `ssl_password_file.t` test fails. To reproduce, do the following:

1. [Download the Prove test suite](http://hg.nginx.org/nginx-tests/archive/tip.tar.gz) from the NGINX web site.
2. Build the module as a shared library (HEAD on `fix/master/31-compile-as-shared-module` ATM)
3. From the test suite, run `TEST_NGINX_GLOBALS="load_module /home/tspencer/nginx_phantom_token_module/nginx-1.13.7/objs/ngx_curity_http_phantom_token_module.so;"   TEST_NGINX_BINARY=$(which nginx) prove -v ./ssl_password_file.t`

It seems related to the config used during the test. What should that be if not the config below and how should the config be distributed in source with the module?

```
tspencer@ubuntu:~/nginx-tests-6d065c9a3b1f$ TEST_NGINX_BINARY=$HOME/nginx-dev/sbin/nginx prove -v ./ssl_password_file.t
./ssl_password_file.t ..
1..5
ok 1 - ssl_password_file works
ok 2 - http
ok 3 - https
ok 4 - no alerts
ok 5 - no sanitizer errors
ok
All tests successful.
Files=1, Tests=5,  0 wallclock secs ( 0.01 usr  0.00 sys +  0.10 cusr  0.01 csys =  0.12 CPU)
Result: PASS
tspencer@ubuntu:~/nginx-tests-6d065c9a3b1f$ TEST_NGINX_GLOBALS="load_module /home/tspencer/nginx_phantom_token_module/nginx-1.13.7/objs/ngx_curity_http_phantom_token_module.so;"   TEST_NGINX_BINARY=$HOME/nginx-dev/sbin/nginx prove -v ./ssl_password_file.t
./ssl_password_file.t ..
1..5
ok 1 - ssl_password_file works
ok 2 - http
ok 3 - https
ok 4 - no alerts
ok 5 - no sanitizer errors
ok
All tests successful.
Files=1, Tests=5,  1 wallclock secs ( 0.02 usr  0.01 sys +  0.15 cusr  0.02 csys =  0.20 CPU)
Result: PASS
tspencer@ubuntu:~/nginx-tests-6d065c9a3b1f$ TEST_NGINX_GLOBALS="load_module /home/tspencer/nginx_phantom_token_module/nginx-1.13.7/objs/ngx_curity_http_phantom_token_module.so;" TEST_NGINX_GLOBALS_HTTP="include $HOME/nginx_phantom_token_module/nginx.conf;"   TEST_NGINX_BINARY=$HOME/nginx-dev/sbin/nginx prove -v ./ssl_password_file.t
./ssl_password_file.t ..
1..5
not ok 1 - ssl_password_file works

#   Failed test 'ssl_password_file works'
#   at ./ssl_password_file.t line 135.
#          got: 'Can't start nginx at lib/Test/Nginx.pm line 345.
# '
#     expected: ''
not ok 2 - http

#   Failed test 'http'
#   at ./ssl_password_file.t line 139.
#                   undef
#     doesn't match '(?^ms:200 OK.*http)'
not ok 3 - https

#   Failed test 'https'
#   at ./ssl_password_file.t line 140.
#                   undef
#     doesn't match '(?^ms:200 OK.*https)'
# Looks like you planned 5 tests but ran 3.
# Looks like you failed 3 tests of 3 run.
Dubious, test returned 3 (wstat 768, 0x300)
Failed 5/5 subtests

Test Summary Report
-------------------
./ssl_password_file.t (Wstat: 768 Tests: 3 Failed: 3)
  Failed tests:  1-3
  Non-zero exit status: 3
  Parse errors: Bad plan.  You planned 5 tests but ran 3.
Files=1, Tests=3,  0 wallclock secs ( 0.02 usr  0.00 sys +  0.09 cusr  0.02 csys =  0.13 CPU)
Result: FAIL
tspencer@ubuntu:~/nginx-tests-6d065c9a3b1f$ cat $HOME/nginx_phantom_token_module/nginx.conf
#load_module /home/tspencer/nginx_phantom_token_module/nginx-1.13.7/objs/ngx_curity_http_phantom_token_module.so;

#events { worker_connections 1024; }

#http {
    server {
        location / {
            #proxy_pass         "http://httpbin.org/";
            phantom_token off;
            phantom_token_client_credential "client_id" "client_secret";
            phantom_token_realm "api";
            phantom_token_scopes "a b c";
            phantom_token_introspection_endpoint curity;
        }

    #    location curity {
     #       proxy_pass "http://localhost:8443/introspection-jwt";

#            proxy_cache my_cache;
 #           proxy_cache_methods POST;
  #          proxy_cache_key $request_body;
   #         proxy_ignore_headers Set-Cookie;
      #  }
    }
#}
```

**Comment by anestos** [comment]
Try adding  the following in the server part of nginx.conf
```
 ssl_certificate_key inherits.key;
 ssl_certificate inherits.crt;
```

**Comment by travisspencer** [comment]
Yes, that fixed that one test but then broke all others when the `inherits.key` and `inherits.crt` weren't present. Seems like we're testing things wrong or the tests are assuming things that don't hold in our case.  

I wrote the following script that only sets those two SSL-related directives when it should:

```bash
#!/bin/sh

export TEST_NGINX_GLOBALS="load_module $HOME/nginx-dev/modules/ngx_curity_http_phantom_token_module.so;"
export TEST_NGINX_BINARY=$HOME/nginx-dev/sbin/nginx
export TEST_NGINX_LEAVE=1
export TEST_NGINX_VERBOSE=1
#export TEST_NGINX_CATLOG=1
export TEST_NGINX_GLOBALS_HTTP="server {
    location /foo {
        phantom_token off;
    }
}"

cd $HOME/nginx-tests-*

if [ $# -eq 0 ]; then
    # Skip the test that requires extra config; this will be tested later.
    TESTS=$(find . -type f -not -name 'ssl_password_file.t' -name "*.t")
else
    TESTS=$*
fi

if [ "$TESTS" != "ssl_password_file.t" ]; then
    # Only run this when the arg isn't the problematic test
    prove $TESTS
fi

if [ $# -eq 0 ] || [ $1 == "ssl_password_file.t" ]; then
    export TEST_NGINX_GLOBALS_HTTP="server {
        ssl_certificate_key inherits.key;
        ssl_certificate inherits.crt;

        location /foo {
            phantom_token off;
        }
    }"

    exec prove ssl_password_file.t
fi
```

It's not perfect, but it works in the common cases, like:

`run_tests.sh`
`run_tests.sh ssl_password_file.t`
`run_tests.sh gzip.t`

**Comment by travisspencer** [comment]
This was fixed in 6df0bacfa4739347c5d463d0222e545a0cd28c90 by allow the enable/disable flag to be present at the HTTP and server level of config as well as the location level.
